tools/ansible/tasks/wsu/test/e2e/wsu_test.go

```diff
@@ -288,3 +298,246 @@ func testHNSNetworksCreated(t *testing.T) {
+	linuxCurlerCommand := []string{"bash", "-c", "yum update; yum install curl -y; curl " + winServerPod.Status.PodIP}
+	linuxCurlerJob, err := createLinuxJob("linux-curler", linuxCurlerCommand)
+	require.NoError(t, err, "Could not create Linux job")
+	defer deleteJob(linuxCurlerJob.Name)
+	err = waitUntilJobSucceeds(linuxCurlerJob.Name)
+	assert.NoError(t, err, "Could not curl the Windows server from a linux container")
+
+	// test Windows <-> Windows on same node
+	// This will continually try to read from the Windows Server. We have to try multiple times as the network is flaky
+	// on container startup
+	winCurlerCommand := []string{"powershell.exe", "-command", "for (($i =0), ($j = 0); $i -lt 10; $i++) { " +
+		"$response = Invoke-Webrequest -UseBasicParsing -Uri " + winServerPod.Status.PodIP +
+		"; $code = $response.StatusCode; echo \"GET returned code $code\";" +
+		"If ($code -eq 200) {exit 0}; Start-Sleep -s 10;}; exit 1" + winServerPod.Status.PodIP}
+	winCurlerJob, err := createWindowsServerJob("win-curler", winCurlerCommand)
+	require.NoError(t, err, "Could not create Windows job")
+	defer deleteJob(winCurlerJob.Name)
+	err = waitUntilJobSucceeds(winCurlerJob.Name)
+	assert.NoError(t, err, "Could not curl the Windows server from a Windows container")
+
+	// TODO: test Windows <-> Windows on different node
+}
+
+// waitUntilJobSucceeds will return an error if the job fails or reaches a timeout
+func waitUntilJobSucceeds(name string) error {
+	var job *batchv1.Job
+	var err error
+	for i := 0; i < 10; i++ {
+		job, err = k8sclientset.BatchV1().Jobs(v1.NamespaceDefault).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if job.Status.Succeeded > 0 {
+			return nil
+		}
+		if job.Status.Failed > 0 {
+			return fmt.Errorf("job %v failed", job)
+		}
+		time.Sleep(5 * time.Second)
+	}
+	return fmt.Errorf("job %v timed out", job)
+}
+
+// waitUntilPodHasRunningContainer will return a pod that matches the selector, once the container specified by the pod
+// is in the "Running" state
+func waitUntilPodHasRunningContainer(selector metav1.LabelSelector) (v1.Pod, error) {
+	var err error
+	pod := v1.Pod{}
+	for i := 0; i < 10; i++ {
+		pod, err = getPodWithLabelSelector(selector)
+		if err != nil {
+			return v1.Pod{}, err
+		}
+		if len(pod.Status.ContainerStatuses) > 1 {
+			return v1.Pod{}, fmt.Errorf("expected one container within the pod but found %d",
+				len(pod.Status.ContainerStatuses))
+		} else if len(pod.Status.ContainerStatuses) == 1 {
+			container := pod.Status.ContainerStatuses[0]
+			if container.State.Running != nil {
+				return pod, nil
+			}
+		}
+		time.Sleep(5 * time.Second)
+	}
+	return pod, fmt.Errorf("timed out waiting for pod %v to have ready container", pod)
+}
+
+// getPodWithLabelSelector returns the pod that matches the label selector. If more than one pod match the selector,
+// the function will return an error
+func getPodWithLabelSelector(selector metav1.LabelSelector) (v1.Pod, error) {
```

Comment:
If you check if the deployment has scaled up then you won't need this function.

Reply:
I still need the function to retrieve the pod IP

Reply:
Then that is the only thing this function should return.